irc/dline.go

```diff
@@ -167,30 +182,32 @@ func (dm *DLineManager) CheckIP(addr net.IP) (isBanned bool, info *IPBanInfo) {
 	}
 
 	// check networks
-	var netsToRemove []net.IPNet
-
-	for _, netInfo := range dm.networks {
-		if !netInfo.Network.Contains(addr) {
-			continue
+	doCleanup := false
+	defer func() {
+		if doCleanup {
+			go func() {
+				dm.Lock()
+				defer dm.Unlock()
+				for key, netInfo := range dm.networks {
+					if netInfo.Info.Time.IsExpired() {
+						delete(dm.networks, key)
```

Comment:
Hmm, will deleting elements while we're iterating through it cause any issues? Same for the `entries` slice below in the kline manager

Reply:
This is apparently safe: https://stackoverflow.com/a/23230406

Reply:
Ah fair enough, awesome! All good in that case.